### ISSUE TITLE
hx-get doesn't include all inputs in a form.

### ISSUE THREAD

**Issue by pranaysashank** [bigskysoftware/htmx#735]
When a hx-get is placed on an input tag in a form, I am expecting htmx to send all other inputs tags in the form as well similar to how hx-post. But this php example demonstrates that it doesn't happen.

```php
<html>
<head>
    <script src="https://unpkg.com/htmx.org@1.6.1"></script>
    <script>htmx.logAll();</script>
</head>
<body>
<div>
    <h1>Test</h1>
    <hr>
    <div id="container">
        <h2>POSTs:</h2>
        <?php
        var_dump($_POST);
        ?>
        <hr>
        <h2>GETs:</h2>
        <?php
        var_dump($_GET);
        ?>
        <hr>
        <h3>POST Form</h3>
        <form method="post" >
            <input type="text" name="post_variable" value=""
                   hx-post="/htmx.php"
                   hx-select="#container"
                   hx-target="#container">
            <input type="text" name="unseen_post_variable"
                   value=""
                   >
            <button type="submit">Submit</button>
        </form>
        <h3>GET Form</h3>
        <form method="get" >
            <input type="text" name="get_variable" value=""
                   hx-get="/htmx.php"
                   hx-select="#container"
                   hx-target="#container">
            <input type="text" name="unseen_get_variable"
                   value=""
                   >
            <button type="submit">Submit</button>
        </form>
    </div>
</div>
</body>
</html>
```

- #### Browser: Firefox

**Comment by Indribell** [comment]
I tried out your code and it works perfectly in Edge (Chromium).

> ?get_variable=&unseen_get_variable=
> ?get_variable=ddd&unseen_get_variable=ddd

The URI shows the correct GET Form information, be it filled or empty. Same with POST Form information in the body, as per HTTP specs.

So your looking at a potential browser or server issue. Maybe auto filtering on empty variables.

**Comment by niceperl** [comment]
I have tried something similar, with a selector in hx-include that search for the values of a subset of inputs fields (3 elements in my test). 
If I use "hx-get", only one parameter (of 3) is included, just the one that triggered the click event.
If I change the method to hx-post, the request payload will include the expected 3 parameters. 

This is the most relevant part of my code (with hx-get):

```
<select class="select-fp" hx-get="/server-call" hx-include="#myform .select-fp" hx-target="#my-div" hx-trigger="click" name="param_1"> ... </select>

<select class="select-fp" hx-get="/server-call" hx-include="#myform .select-fp" hx-target="#my-div" hx-trigger="click" name="param_2"> ... </select>

<select class="select-fp" hx-get="/server-call" hx-include="#myform .select-fp" hx-target="#my-div" hx-trigger="click" name="param_3"> ... </select>
```

Thanks.

**Comment by David-Guillot** [comment]
Hi there!

@pranaysashank Not sure if this issue is still relevant to you, but if it is, clearly [hx-include](https://htmx.org/attributes/hx-include/)  is what you need. The [automatic/implicit/opiniated include of the closest form for non-GET requests](https://github.com/bigskysoftware/htmx/blob/master/src/htmx.js#L2074)  could probably be more documented cc @1cg 😉 

@niceperl your comment intrigues me ; have you tried this code with htmx 1.7.x? Have you tried a wider selector, like selecting a parent element of all three inputs?

**Comment by pranaysashank** [comment]
If it's the intended behaviour then it's fine I guess, just need documentation for it. Feel free to close this issue

**Comment by niceperl** [comment]
@David-Guillot I have tested it with 1.6.1 version. As soon as I found a solution I haven't tried it again in newer releases of htmx

**Comment by 1cg** [comment]
Yes, we don't include an enclosing forms values for GET, as documented here: https://htmx.org/docs/#parameters

This is to prevent blowing out the address bar on things like boosted link clicks etc, although you can still do so explicitly w/ `hx-include`
